canary/producer-cpp/src/Canary.cpp

```diff
@@ -303,226 +156,152 @@ VOID pushErrorMetrics(CustomData *cusData, double duration)
-        metricDatum, cusData->pDimensionPerStream, cwRequest);
+               metricDatum, cusData->pDimensionPerStream, cwRequest);
     LOG_DEBUG("Current View Duration: " << currentViewDuration);
 
     if (cusData->pCanaryConfig->useAggMetrics)
     {
         pushMetric("FrameRate", frameRate, Aws::CloudWatch::Model::StandardUnit::Count_Second,
-            metricDatum, cusData->pAggregatedDimension, cwRequest);
+                   metricDatum, cusData->pAggregatedDimension, cwRequest);
         pushMetric("TransferRate", transferRate, Aws::CloudWatch::Model::StandardUnit::Kilobits_Second,
-            metricDatum, cusData->pAggregatedDimension, cwRequest);
+                   metricDatum, cusData->pAggregatedDimension, cwRequest);
         pushMetric("CurrentViewDuration", currentViewDuration, Aws::CloudWatch::Model::StandardUnit::Milliseconds,
-            metricDatum, cusData->pAggregatedDimension, cwRequest);
+                   metricDatum, cusData->pAggregatedDimension, cwRequest);
     }
 
     // Send metrics to CW
     cusData->pCWclient->PutMetricDataAsync(cwRequest, onPutMetricDataResponseReceivedHandler);
 }
 
-VOID pushStartupLatencyMetric(CustomData *data)
+// put frame function to publish metrics to cloudwatch after getting g signal from producer sdk cpp
+static VOID put_frame_kvs(GstElement *kvsSink, VOID *gMetrics, gpointer data)
 {
-    double currentTimestamp = duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count();
-    double startUpLatency = (double)(currentTimestamp - data->startTime / 1000000); // [milliseconds]
-    Aws::CloudWatch::Model::MetricDatum startupLatencyDatum;
-    Aws::CloudWatch::Model::PutMetricDataRequest cwRequest;
-    cwRequest.SetNamespace("KinesisVideoSDKCanary");
-
-    LOG_DEBUG("Startup Latency: " << startUpLatency);
-
-    pushMetric("StartupLatency", startUpLatency, Aws::CloudWatch::Model::StandardUnit::Milliseconds, startupLatencyDatum, data->pDimensionPerStream, cwRequest);
-    if (data->pCanaryConfig->useAggMetrics)
+    LOG_DEBUG("put frame at canary");
+    CustomData *cusData = (CustomData*) data;
+    KvsSinkMetric *kvsSinkMetric = reinterpret_cast<KvsSinkMetric *> (gMetrics);
+    updateFragmentEndTimes(kvsSinkMetric->framePTS, cusData->lastKeyFrameTime, cusData->timeOfNextKeyFrame);
+    pushStreamMetrics(cusData, kvsSinkMetric->streamMetrics);
+    pushClientMetrics(cusData, kvsSinkMetric->clientMetrics);
+
+    double duration = duration_cast<seconds>(system_clock::now().time_since_epoch()).count() - cusData->timeCounter;
+    // Push error metrics and logs every 60 seconds
+    if(duration > 60)
     {
-        pushMetric("StartupLatency", startUpLatency, Aws::CloudWatch::Model::StandardUnit::Milliseconds, startupLatencyDatum, data->pAggregatedDimension, cwRequest);
+        pushErrorMetrics(cusData, duration, kvsSinkMetric->streamMetrics);
+        cusData->pCanaryLogs->canaryStreamSendLogs(cusData->pCloudwatchLogsObject);
```

Comment:
Why do we need sendLogs? We are not sending any logs. Remove everything related to logs